src/pybind/ceph_volume_client.py

```diff
@@ -1341,3 +1342,60 @@ def clone_volume_to_existing(self, dest_volume_path, src_volume_path, src_snapsh
         src_snapshot_path = self._snapshot_path(self._get_path(src_volume_path), src_snapshot_name)
 
         self._cp_r(src_snapshot_path, dest_fs_path)
+
+    def put_object(self, pool_name, object_name, data):
+        """
+        Synchronously write data to an object.
+
+        :param pool_name: name of the pool
+        :type pool_name: str
+        :param object_name: name of the object
+        :type object_name: str
+        :param data: data to write
+        :type data: bytes
+        """
+        ioctx = self.rados.open_ioctx(pool_name)
+        # Restrict write size to 4MiB.
+        max_size = 4096 * 1024
```

Comment:
Might be simpler to just use the rados module's `conf_get` method to fetch `osd_max_write_size` here if you just need a "sanity" limit -- that's the limit that the OSD imposes on how much data can be written in a single operation in MB (default is 90).

Reply:
OK. So I guess I can also read osd_max_write_size 90 MB in a single rados read call while fetching the object?

I looked at the rados CLI 'get' and 'put' implementation in src/tools/rados/rados.cc. Heavily simplifying, for 'put' it writes 'op_size', and for 'get' reads 'op_size' at a time, and the default 'op_size' is 4MB.  So I thought it's a reasonable size limit to read and write config data atomically without need for locking.

Reply:
> So I guess I can also read osd_max_write_size 90 MB in a single rados read call while fetching the object?

Yep!

Not that 90MB objects are necessarily smart, it's just there as a sanity/safety limit to protect the OSD from people sending gratuitously oversized operations.

I guess it depends on the purpose of the 4MB limit here -- if the purpose is sanity/safety, then I would just rely on the underlying OSD limit.  However, if we want not just safety but to ensure people are using this interface only for "small" things like config files, then a much lower limit like 4MB is fine.

I'm probably making too much of an issue out of this anyway -- there's nothing wrong with just having this code take an opinionated stance and apply the 4MB limit.

Reply:
Yeah, I should've been explicit about what I was going for with the size limit. The 'osd_max_write_size' limit seems more intuitive to me.